### ISSUE TITLE
Telegram as OAuth providers

### ISSUE THREAD

**Issue by hsluoyz** [casdoor/casdoor#1363]
### Discussed in https://github.com/casdoor/casdoor/discussions/1358

<div type='discussions-op-text'>

<sup>Originally posted by **Vladimir-Va** December  4, 2022</sup>
Hello is possible to add [Telegram messenger](https://web.telegram.org/) as OAuth providers?
https://core.telegram.org/api/auth#sign-insign-up</div>

Proposed by: [Vladimir-Va](https://github.com/Vladimir-Va)

## Update

We should add all providers here: https://github.com/markbates/goth#supported-providers to Casdoor

**Comment by forestmgy** [comment]
After researching, I think it is a bit difficult for casdoor to support telegram as OAuth provider. 
There are two reasons:
1. Telegram does not support classic OAuth login. I googled about telegram's oauth information and found two issues. 
One is to discuss adding telegram as an oauth provider to the nextauthjs [link here.](https://github.com/nextauthjs/next-auth/issues/2406)
Another is to request the telegram team to add support for OAuth to telegram. It can be seen that telegram itself does not support classic OAuth [link here.](https://bugs.telegram.org/c/23608)
2. Telegram supports a login method that directly adds a button on the web page, https://core.telegram.org/widgets/login
This login method is very rigid, and can only be set according to the style given by telegram, and it is impossible to set the icon style like casdoor.
To sum up, I think adding telegram to OAuth provider is not a good option now.

**Comment by mserico** [comment]
replaced by: https://github.com/casdoor/casdoor/issues/4569
